### ISSUE TITLE
Extra field for Connector: boolean neuter (default: true)

### ISSUE THREAD

**Issue by simonphilips** [ocpi/ocpi#24]
Most power nets with 3 phases have a neuter. The voltage between neuter and phase 1 is about 230V and between two phases 400V (at least in Europe). However, there are some old nets with 3 phases that do not have a neuter (large parts in Brussels and I think Berlin and Norway as well). Type 2 connectors work, but the N part is simply not connected (or connected to the grounding). In those cases the voltage between two phases is only 230V instead of 400V.

I think it's important to be able to pass this information along as some cars (Tesla) will detect an error and refuse to charge on such connectors. As most of Europe has a proper net, I suggest making this field optional and true by default.

**Comment by poohsen** [comment]
Sounds like you could turn this into a pull request right away.

**Comment by LeonHuijsdens** [comment]
Hi Simon, is this a connector-field then, or is it an attribute for the charging station as a whole?

**Comment by simonphilips** [comment]
Good point. It's for the whole charging location. If there is no neuter in the street, then you can't magically make one appear. And when there is a neuter, it really does not make sense to not connect it to the charger. So it should be an attribute of the charging location indeed.

**Comment by simonphilips** [comment]
Unfortunately, it's more complicated than I expected. I'm not sure how far we should go with this.

I've taken a photo from a brochure we have that explains the situation a bit:
https://my.becharged.eu/media/ocpi/powergrids.jpg

To summarise, there are several types of power grids here (and elsewhere in Europe, though not standard):
* 3 x 400V+N : always works
* 3 x 230V (star) [no neutral] : some cars can charge, others can't
* 3 x 230V (triangle) [no neutral] : some cars can charge, others can't
* (single phase) 1 x 230V : slow charging, usually domestic chargers

I just don't know how the voltage should be communicated in the protocol:
* 3 x 400V+N: line-to-line = 400V; line-to-neutral = 230V; line-to-earth = 230V
* 3 x 230V (star): line-to-line = 230V; line-to-neutral = doesn't exist; line-to-earth = 133V
* 3 x 230V (triangle): line-to-line = 230V; line-to-neutral = doesn't exist; line-to-earth = 230V (phase 1&2)  / 0V (phase 3)
* (single phase) 1 x 230V: line-to-line = doesn't exist; line-to-neutral = 230V; line-to-earth = 230V

Each type of voltage has one scenario where it doesn't exist (or is not uniquely defined).

The easiest solution would be to define the voltage as "line-to-line" for 3 phases and as "line-to-neutral" for 1 phase systems. This is how we do it in our system now and it is also how Tesla checks for errors: for 3 phases it requires a line-to-line of 400V, for single phase it requires a line-to-neutral of 230V.

But then the 'meaning' of voltage depends on the number of phases. I'm not sure I like that in a protocol, but I don't see good alternatives either.

**Comment by poohsen** [comment]
I'd say it depends on the prevalence of those types of networks. 

How about making voltage  mean line-to-line by default and maybe have an optional N-voltage field for line-to-neutral for 1 phase? It would be nice if operators didn't have to bother with this if they don't have those kind of grid connections in their network.

This *could* be combined with restructuring the power field from this:
* power (replaces enum of DC50kWh / AC11kWh etc etc)
 * current (ac_1_phase, ac_2_phase, ac_3_phase, dc)
 * voltage
 * amperage

where `current` is a list, to a list of triplets:
`[
   {current: "ac_1_phase", n-voltage: "230V", amperage: 16}, 
   {current: "ac_3_phase", voltage: "400V", amperage: 16}
]`

with the constraint that either voltage or n-voltage has to be present in a triple. 
It would avoid giving `voltage` ambiguous semantics.
Operators not having

**Comment by simonphilips** [comment]
Yeah, that might be a good solution as well. It avoids the double definition though it might make dealing with it from a technical point of view a bit annoying. I'm not sure what the right answer is. OCPP does this, but the 2.0 draft mentions that it might disappear.

These (older) types of networks are common enough in Belgium (and elsewhere in Europe) to ignore them unfortunately, and it is important to distinguish them from the 3x400V+N as some brands of cars will check for the 400V between phases (such as the Tesla).

Unrelated remark: we can remove the ac_2_phase option as 2 phased systems do not exist.

**Comment by simonschllng** [comment]
### Grid Types
There are common grid type names in electrical engineering. Those describe how the transformer is connected to the ground. The grid type is independent of the voltage. The general types are:
* TT
* TN
* IT

TN has sub-types, which are named TN-C, TN-S and TN-C-S. Additionally there is either the star-point grounded or one of the three lines.
In UK, US and some older grids in europe, there is also the split-phase system where you have a single-phase grid.

### Voltage
It is common to use the line-to-line voltage as nominal voltage in a 3-phase grid. However, it depends on how the connector is wired and which one you want to use. See below.

## Usage in the protocol
### Grid type
As the grid type is the same for the whole station, it should be specified on this level. Proposal would be the following list of choices for the grid system:
* TT
* TN-C
* TN-S
* TN-C-S
* TN
* IT
* split-phase
* single-phase

And an additional value for the earthing:
* neutral-earthed
* phase-earthed
* unearthed

### Voltage
If we give this information on socket level, the nominal voltage of the socket should be specified. A single-phase socket like a Type-1 should then be specified as phase-to-earth or phase-to-phase, depending in the way it is connected.

This should cover all possibilities. Some combinations might not be possible and some might be a bit redundant. But not all operators will have all information present. This solution would give the posibility to specify the situation as goog as it is known. Navigation systems in the car, which know how the car is behaving can then filter the spots which are not working.

**Comment by RobertdeLeeuw-IHomer** [comment]
Is this still an issue?
